cloudslang-compiler/src/main/java/io/cloudslang/lang/compiler/modeller/ExecutableBuilder.java

```diff
@@ -307,6 +309,31 @@ private Workflow getOnFailureWorkflow(List<Map<String, Map<String, Object>>> wor
         return onFailureWorkFlow;
     }
 
+    private void handleOnFailureStepNavigationSection(List<Map<String, Map<String, Object>>> onFailureData, String execName, List<RuntimeException> errors) {
+        Map.Entry<String, Map<String, Object>> onFailureStep = getFirstOnFailureStep(onFailureData);
+        if (onFailureStep.getValue().containsKey(NAVIGATION_KEY)) {
+            errors.add(new RuntimeException("Flow: '" + execName + "' syntax is illegal.\nThe step below 'on_failure' property should not contain a \"navigate\" section."));
+        }
+        List<Map<String, String>> failureNavigationSection = getOnFailureNavigationSection();
+        onFailureStep.getValue().put(NAVIGATION_KEY, failureNavigationSection);
+    }
+
+    private List<Map<String, String>> getOnFailureNavigationSection() {
+        List<Map<String, String>> failureNavigationSection = new ArrayList<>();
+        Map<String, String> success = new HashMap<>();
+        success.put(ScoreLangConstants.SUCCESS_RESULT, ScoreLangConstants.FAILURE_RESULT);
+        Map<String, String> failure = new HashMap<>();
+        failure.put(ScoreLangConstants.FAILURE_RESULT, ScoreLangConstants.FAILURE_RESULT);
```

Comment:
I think you should not put any navigation here, since you will set the actual results later on in socrecompiler (and the subflow may not have these results)